src/publications_embedding.py

```diff
@@ -0,0 +1,131 @@
+    """
+    Get the embeddings of the publications for every author
+
+    Args:
+        pubs (list[str]):
+        embedding_name (str): The name of the embedding model. By default the
+            model text-embedding-ada-002 is used.
+        max_token (int): The maximum number of tokens for which an embedding is
+            computed. By default this is the maximum number of tokens of the 
+            embedding model text-embedding-ada-002.
+        
+    Returns:
+        embeddings (numpy.ndarray): Embeddings of the publication lists.
+    """
+
+    embedding = []
+
+    for pub in pubs:
+        embedding.append(embedding_from_string(pub,
+                                               embedding_name = embedding_name,
+                                               max_token = max_token))
+        if embedding[-1] == None:
+            print("ERROR: The embedding for \"", pub, "\" could not be" 
+                  + "computed! Please check your input and parameters!")
+            exit(1)
+
+    embedding = np.asarray(embedding)
+
+    return embedding
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Compute embeddings for lists of publications.')
+    parser.add_argument('-p', '--publications', type = try_to_read_dir, 
+                        required = True,
+                        help = 'Directory containing files with names ' 
+                        + '<author_id>.txt, which contain publication lists.')
+    parser.add_argument('-n', '--num_authors', type = int, required = True,
+                        help = 'Number of expected files in the publications ' 
+                        + 'directory, i.e. if the highest author ID in the ' 
+                        + 'directory is 10, specify 11 as the indexing starts '
+                        + 'with 0. All files in {0.txt, ... n-1.txt} are ' 
+                        + 'tried, a subset may be missing.')
+    parser.add_argument('-o', '--outfile', default = 'pub_embed.h5',
+                        help = 'Output file in hdf5 format with data sets: ' 
+                        + 'publication_embedding: 2-dim numpy array with '
+                        + 'embeddings. author_ids: 1-dim numpy array '
+                        + 'containing the ids for each embedding.')
+    args = parser.parse_args()        
+
+    # read publications and compute embeddings
+    pubs, author_ids = read_pubs(args.publications, args.num_authors)
+    embeddings = embeddings_from_pubs(pubs)
+
+    # write data to hdf5 file
+    with h5py.File(args.outfile, 'w') as f_out:
+        f_out.create_dataset(name = 'publication_embedding', 
+                             data = embeddings,
+                             compression = 'gzip')
+        f_out.create_dataset(name = 'author_ids', 
+                             data = author_ids,
+                             compression = 'gzip')
+    
+    exit(0)
+
+
+if __name__ == "__main__":
+    main()
+    
```

Comment:
There is still an EOF without EOL. I can't seem to do this with suggestions. You have to do it. You can see this with "compare" on the branch and then see whether there is a red circled - sign at the end of the file.